### PR TITLE
[IDE-2] arduino ide2 security doc

### PR DESCRIPTION
**Pull request by rhaidiz**
## What This PR Changes
Add an article describing the security measures taken to ensure secure development of the Arduino IDE2.

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

### CODE REVIEW

**Comment by stevisco** [comment]
the title of the article shall be "Security of Arduino IDE" and it shall be linked in both IDE1 and IDE2 list of articles (for IDE1 users, it contains recommendation to update)

**Comment by rhaidiz** [comment]
I applied the changes, you can review them in the deploy review.

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+---
+title: 'Security of Arduino IDE'
+description: 'Learn the secure development process behind Arduino IDE.'
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:3]:
```suggestion
description: 'Learn about the secure development process behind the Arduino IDE.'

```

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+tags:
+ - Security
+author: 'Arduino Security Team'
+---
+
+Arduino IDE 2.x is the latest version of the Arduino Programming tool and it is built on Eclipse Theia, an open-source framework for building IDEs.
+
+Arduino is committed to regularly monitor and update security measures applied to the Arduino IDE to ensure proper protection from any threats or vulnerabilities detected.
+
+Arduino defined principles and requirements to be followed within the secure development lifecycle of Hardware, System and Software and the development of the open-source Arduino software makes no exception. In particular, Arduino follows the **Secure by Design** principle in every stage of the software development and the **Security Principles** listed below are followed during the secure development lifecycle:
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:13]:
```suggestion
Arduino's defined principles and requirements to be followed within the secure development lifecycle of Hardware, System and Software and the development of the open-source Arduino software makes no exception. In particular, Arduino follows the **Secure by Design** principle in every stage of the software development and the **Security Principles** listed below are followed during the secure development lifecycle:

```

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- **Secure Repository Management**: with the purpose of guaranteeing that the Least of privilege and Separation of duties principles are in place to protect the software repositories.
+- **Secure Environment**: in order to ensure the adequate segregation of the services and data contained into Development / Staging / Production environments.
+- **Secure Development**: with the purpose of guaranteeing that security risks, security requirements and best practices are taken into consideration in all the steps of the development lifecycle for Hardware, Software and Infrastructure artifacts and reducing the residual risk associated with security flaws and bugs managed through the internal Vulnerability Management Program.
+- **Security Testing**: in order to ensure that all required security activities such as, but not limited to: Design Review, Threat and Risk Analysis, Code Review, Penetration Test, Vulnerability Assessment and all fixing activities related to the Vulnerability Management Program are performed by the security team, with the help of the developers, along all stages of the development lifecycle.
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:31]: @rhaidiz if possible, divide up these sentences as they are quite long (around 20 words per sentence is ideal).

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+As part of the security testing activities, Arduino periodically performs the following tasks on the Arduino IDE:
+
+- **Secure Code Review (SCR)**: the code is reviewed in order to ensure the safety against main security vulnerabilities and threats.
+- **Secure Component Analysis (SCA)**: security engineers evaluate components related to Theia npm packages and Electron.
+- **Secrets scanning analysis**: activities performed to ensure that secrets are not inadvertently leaked within the repository.
+
+Finally, should an Arduino user or customer suspect a vulnerability or security issue, they are invited to report it as described in our Coordinated Vulnerability Disclosure policy available at: [https://www.arduino.cc/en/security](https://www.arduino.cc/en/security).
+
+## Third party components
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:41]:
```suggestion
## Third Party Components

```

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+In the process of conducting Secure Component Analysis, Arduino puts particular attention on the aforementioned  external dependencies (Eclipse Theia and OpenJS Electron) and reports any found vulnerability to the respective project maintainer:
+
+- Eclipse Theia manages vulnerabilities using the process described in [https://www.eclipse.org/security](https://www.eclipse.org/security).
+- OpenJS Electron vulnerabilities are managed using the process described in [https://www.electronjs.org/docs/latest/tutorial/security](https://www.electronjs.org/docs/latest/tutorial/security).
+
+When vulnerabilities in third party components are fixed by the respective maintainer, Arduino will update the component involved as required, in a commercially reasonable time, based on the severity of the identified vulnerability.
+
+## Versions management
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:49]:
```suggestion
## Version Management

```

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
+---
+title: 'Security of Arduino IDE'
+description: 'Learn the secure development process behind Arduino IDE.'
+tags:
+ - Security
+author: 'Arduino Security Team'
+---
+
+Arduino IDE 2.x is the latest version of the Arduino Programming tool and it is built on Eclipse Theia, an open-source framework for building IDEs.
```
Comment (karlsoderby) [inline content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:9]: This needs to change to IDE 1.8.x (or legacy IDE).

File: content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+tags:
+ - Security
+author: 'Arduino Security Team'
+---
+
+Arduino IDE 2.x is the latest version of the Arduino Programming tool and it is built on Eclipse Theia, an open-source framework for building IDEs.
+
+Arduino is committed to regularly monitor and update security measures applied to the Arduino IDE to ensure proper protection from any threats or vulnerabilities detected.
+
+Arduino defined principles and requirements to be followed within the secure development lifecycle of Hardware, System and Software and the development of the open-source Arduino software makes no exception. In particular, Arduino follows the **Secure by Design** principle in every stage of the software development and the **Security Principles** listed below are followed during the secure development lifecycle:
```
Comment (karlsoderby) [inline content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md:13]:
```suggestion
Arduino's defined principles and requirements to be followed within the secure development lifecycle of Hardware, System and Software and the development of the open-source Arduino software makes no exception. In particular, Arduino follows the **Secure by Design** principle in every stage of the software development and the **Security Principles** listed below are followed during the secure development lifecycle:

```

File: content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+- **Secure Repository Management**: with the purpose of guaranteeing that the Least of privilege and Separation of duties principles are in place to protect the software repositories.
+- **Secure Environment**: in order to ensure the adequate segregation of the services and data contained into Development / Staging / Production environments.
+- **Secure Development**: with the purpose of guaranteeing that security risks, security requirements and best practices are taken into consideration in all the steps of the development lifecycle for Hardware, Software and Infrastructure artifacts and reducing the residual risk associated with security flaws and bugs managed through the internal Vulnerability Management Program.
+- **Security Testing**: in order to ensure that all required security activities such as, but not limited to: Design Review, Threat and Risk Analysis, Code Review, Penetration Test, Vulnerability Assessment and all fixing activities related to the Vulnerability Management Program are performed by the security team, with the help of the developers, along all stages of the development lifecycle.
```
Comment (karlsoderby) [inline content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md:31]: Same here, divide up sentences to around 20 words for better readability.

File: content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+As part of the security testing activities, Arduino periodically performs the following tasks on the Arduino IDE:
+
+- **Secure Code Review (SCR)**: the code is reviewed in order to ensure the safety against main security vulnerabilities and threats.
+- **Secure Component Analysis (SCA)**: security engineers evaluate components related to Theia npm packages and Electron.
+- **Secrets scanning analysis**: activities performed to ensure that secrets are not inadvertently leaked within the repository.
+
+Finally, should an Arduino user or customer suspect a vulnerability or security issue, they are invited to report it as described in our Coordinated Vulnerability Disclosure policy available at: [https://www.arduino.cc/en/security](https://www.arduino.cc/en/security).
+
+## Third party components
```
Comment (karlsoderby) [inline content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md:41]:
```suggestion
## Third Party Components

```

File: content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+In the process of conducting Secure Component Analysis, Arduino puts particular attention on the aforementioned  external dependencies (Eclipse Theia and OpenJS Electron) and reports any found vulnerability to the respective project maintainer:
+
+- Eclipse Theia manages vulnerabilities using the process described in [https://www.eclipse.org/security](https://www.eclipse.org/security).
+- OpenJS Electron vulnerabilities are managed using the process described in [https://www.electronjs.org/docs/latest/tutorial/security](https://www.electronjs.org/docs/latest/tutorial/security).
+
+When vulnerabilities in third party components are fixed by the respective maintainer, Arduino will update the component involved as required, in a commercially reasonable time, based on the severity of the identified vulnerability.
+
+## Versions management
```
Comment (karlsoderby) [inline content/software/ide-v2/tutorials/ide-v2-security/ide-v2-security.md:49]:
```suggestion
## Version Management

```

**Review by karlsoderby — CHANGES_REQUESTED** [review]
Hey @rhaidiz , article looks really good. Left a couple of suggestions and a few very minor issues to resolve before we can merge.

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md (hunk 6 above)
Comment (rhaidiz) [reply content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:10]: I know this page is under the IDE v1, however the content of what comes right afterwards refers to the IDE v2. The main purpose for this article for being under the IDE v1 I think is within section "Version Management".

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md (hunk 3 above)
Comment (rhaidiz) [reply content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:33]: I very much agree with you, all these sentences should be reworked, but I ultimately decided to keep them like this to be consistent with what we have posted for the Arduino [Cloud Security Considerations](https://docs.arduino.cc/arduino-cloud/features/security-considerations). However, I would be more than happy to rework the sentences both here and in the Arduino Cloud Security article. If @stevisco also agrees I'll do that here and finally I'll align the other document accordingly.

File: content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md (hunk 6 above)
Comment (stevisco) [reply content/software/ide-v1/tutorials/ide-v1-security/ide-v1-security.md:10]: Federico is correct, the sentence is and remains about IDE2 even if the article is inside IDE1 page. We want IDE1 users to understand that they need to move to IDE2 which is the latest and maintained. see the story on version management in this page

**Review by karlsoderby — APPROVED** [review]
LGTM!

**Pull request merged**